packages/teams-js/src/internal/videoFrameTick.ts

```diff
@@ -0,0 +1,50 @@
+import { generateGUID } from './utils';
+
+export class VideoFrameTick {
+  private static readonly setTimeoutCallbacks: {
+    [key: string]: {
+      callback: () => void;
+      startedAtInMs: number;
+      timeoutInMs: number;
+    };
+  } = {};
+
+  public static setTimeout(callback: () => void, timeoutInMs: number): string {
+    const startedAtInMs = performance.now();
+    const id = generateGUID();
+    VideoFrameTick.setTimeoutCallbacks[id] = {
+      callback,
+      timeoutInMs,
+      startedAtInMs,
+    };
+    return id;
+  }
+
+  public static clearTimeout(id: string): void {
+    delete VideoFrameTick.setTimeoutCallbacks[id];
+  }
+
+  public static setInterval(callback: () => void, intervalInMs: number): void {
+    VideoFrameTick.setTimeout(function next() {
+      callback();
+      VideoFrameTick.setTimeout(next, intervalInMs);
+    }, intervalInMs);
+  }
+
+  public static tick(): void {
+    const now = performance.now();
+    const timeoutIds = [];
+    for (const key in VideoFrameTick.setTimeoutCallbacks) {
+      const callback = VideoFrameTick.setTimeoutCallbacks[key];
+      const start = callback.startedAtInMs;
+      if (now - start >= callback.timeoutInMs) {
+        timeoutIds.push(key);
+      }
+    }
+    for (const id of timeoutIds) {
+      const callback = VideoFrameTick.setTimeoutCallbacks[id];
+      callback.callback();
+      delete VideoFrameTick.setTimeoutCallbacks[id];
+    }
```

Comment:
I feel like comments would help understand exactly what is happening here.

Reply:
Updated, added comments